File: leetCode/双指针/283移动0.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let length = nums.length;
    let p_left = 0;
    let p_right = 0;
    while(p_right < length) {
        if (nums[p_right]) {
            let temp = nums[p_right]
            nums[p_right] = nums[p_left]
            nums[p_left] = temp
            p_left++
        }
        p_right++
    }
};

File: leetCode/2.双指针/283移动0.go
import "fmt"

func swap(nums []int, left int, right int) {
	temp := nums[left]
	nums[left] = nums[right]
	nums[right] = temp
}
func moveZeroes(nums []int) {
	p_left := 0
	p_right := 0
	for p_right < len(nums) {
		if nums[p_right] != 0 {
			swap(nums, p_left, p_right)
			p_left++
		}
		p_right++
	}
}
